rest-api/resources/src/main/java/org/eclipse/kapua/app/api/resources/v1/resources/UsersMfa.java

```diff
@@ -0,0 +1,126 @@
+
+    @Inject
+    public MfaOptionService mfaOptionService;
+
+    /**
+     * Creates a new {@link MfaOption} for the user specified by the "userId" path parameter.
+     *
+     * @param scopeId The {@link ScopeId} in which to create the {@link MfaOption}
+     * @param userId  The {@link EntityId} of the User to which the {@link MfaOption} belongs
+     * @return The newly created {@link MfaOption} object.
+     * @throws KapuaException Whenever something bad happens. See specific {@link KapuaService} exceptions.
+     * @since 1.4.0
+     * @deprecated since 2.0.0 - use POST {scopeId}/user/mfa instead (see {@link UserMfa})
+     */
+    @POST
+    @Path("{userId}/mfa")
+    @Consumes({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    @Deprecated
+    public Response createMfa(
+            @PathParam("scopeId") ScopeId scopeId,
+            @PathParam("userId") EntityId userId) throws KapuaException {
+        return returnCreated(mfaOptionService.create(new MfaOptionCreatorImpl(scopeId, userId)));
+    }
+
+    /**
+     * Returns the {@link MfaOption} of the user specified by the "userId" path parameter.
+     *
+     * @param scopeId The {@link ScopeId} of the requested {@link MfaOption}
+     * @param userId  The {@link EntityId} of the User to which the {@link MfaOption} belongs
+     * @return The requested {@link MfaOption} object.
+     * @throws KapuaException Whenever something bad happens. See specific {@link KapuaService} exceptions.
+     * @since 1.4.0
+     */
+    @GET
+    @Path("{userId}/mfa")
+    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    public MfaOption findMfa(
+            @PathParam("scopeId") ScopeId scopeId,
+            @PathParam("userId") EntityId userId) throws KapuaException {
+        MfaOption mfaOption = mfaOptionService.findByUserId(scopeId, userId);
+        if (mfaOption == null) {
+            throw new KapuaEntityNotFoundException(MfaOption.TYPE, "MfaOption");  // TODO: not sure "MfaOption" it's the best value to return here
+        }
+
+        return mfaOption;
+    }
+
+    /**
+     * Deletes the {@link MfaOption} of the user specified by the "userId" path parameter.
+     *
+     * @param scopeId The {@link ScopeId} of the requested {@link MfaOption}
+     * @param userId  The {@link EntityId} of the User to which the {@link MfaOption} belongs
+     * @return HTTP 200 if operation has completed successfully.
+     * @throws KapuaException Whenever something bad happens. See specific {@link KapuaService} exceptions.
+     * @since 1.4.0
+     */
+    @DELETE
+    @Path("{userId}/mfa")
+    public Response deleteMfa(
+            @PathParam("scopeId") ScopeId scopeId,
+            @PathParam("userId") EntityId userId) throws KapuaException {
+        mfaOptionService.deleteByUserId(scopeId, KapuaSecurityUtils.getSession().getUserId());
```

Comment:
Tested and this is not working. 
It is using the user from the session and not the given one!